src/go/k8s/internal/client/spec.go

```diff
@@ -0,0 +1,99 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"time"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+// KafkaForSpec returns a simple kgo.Client able to communicate with the given cluster specified via KafkaAPISpec.
+func (c *clientFactory) kafkaForSpec(ctx context.Context, namespace string, metricNamespace *string, spec *redpandav1alpha2.KafkaAPISpec, opts ...kgo.Opt) (*kgo.Client, error) {
```

Comment:
Pretty much all of this logic is just what used to be in `kgo-config-options.go`